cloudflare/resource_cloudflare_access_group.go

```diff
@@ -197,38 +197,105 @@ func resourceCloudflareAccessGroupImport(d *schema.ResourceData, meta interface{
 func appendConditionalAccessGroupFields(group cloudflare.AccessGroup, d *schema.ResourceData) cloudflare.AccessGroup {
 	exclude := d.Get("exclude").([]interface{})
 	for _, value := range exclude {
-		group.Exclude = buildAccessGroupCondition(value.(map[string]interface{}))
+		group.Exclude = BuildAccessGroupCondition(value.(map[string]interface{}))
 	}
 
 	require := d.Get("require").([]interface{})
 	for _, value := range require {
-		group.Require = buildAccessGroupCondition(value.(map[string]interface{}))
+		group.Require = BuildAccessGroupCondition(value.(map[string]interface{}))
 	}
 
 	include := d.Get("include").([]interface{})
 	for _, value := range include {
-		group.Include = buildAccessGroupCondition(value.(map[string]interface{}))
+		group.Include = BuildAccessGroupCondition(value.(map[string]interface{}))
 	}
 
 	return group
 }
 
-// buildAccessGroupCondition iterates the provided `map` of values and
+// BuildAccessGroupCondition iterates the provided `map` of values and
 // generates the required (repetitive) structs.
 //
-// Returns the intended combination structure of AccessGroupEmail,
-// AccessGroupEmailDomain, AccessGroupIP and AccessGroupEveryone
-// structs.
-func buildAccessGroupCondition(options map[string]interface{}) []interface{} {
+// Returns the intended combination structure of Access Groups to build the
+// desired policy.
+func BuildAccessGroupCondition(options map[string]interface{}) []interface{} {
 	var group []interface{}
 	for accessGroupType, values := range options {
-		// Since AccessGroupEveryone is a single boolean, we don't need to
-		// iterate over the values like the others so we treat it a little differently.
 		if accessGroupType == "everyone" {
 			if values == true {
-				log.Printf("[DEBUG] values for everyone %s", values)
 				group = append(group, cloudflare.AccessGroupEveryone{})
 			}
+		} else if accessGroupType == "any_valid_service_token" {
+			if values == true {
+				group = append(group, cloudflare.AccessGroupAnyValidServiceToken{})
+			}
+		} else if accessGroupType == "certificate" {
+			if values == true {
+				group = append(group, cloudflare.AccessGroupCertificate{})
+			}
+		} else if accessGroupType == "common_name" {
+			if values != "" {
+				group = append(group, cloudflare.AccessGroupCertificateCommonName{CommonName: struct {
+					CommonName string `json:"common_name"`
+				}{CommonName: values.(string)}})
+			}
+		} else if accessGroupType == "gsuite" {
```

Comment:
This is quite the mess. I'm going to look at cleaning this giant block of conditionals up in another PR.